services/templates/service.go

```diff
@@ -112,3 +115,25 @@ func (s service) UpdateTemplate(c context.Context, template *entities.Template)
 
 	return nil
 }
+
+// DeleteTemplate deletes the given template
+func (s service) DeleteTemplate(c context.Context, template *entities.Template) error {
+	obj, err := s.s3.DeleteObject(&s3.DeleteObjectInput{
+		Bucket: aws.String(os.Getenv("TEMPLATE_BUCKET")),
+		Key:    aws.String(fmt.Sprintf("%d/%d", template.UserID, template.ID)),
+	})
+	if err != nil {
+		return fmt.Errorf("delete object: %w", err)
+	}
+
+	if !aws.BoolValue(obj.DeleteMarker) {
```

Comment:
We don't strictly care about this, if the object does not exist and there is no error, no harm is done. It's better if we continue instead of throwing an error.